harmony/client.py

```diff
@@ -940,10 +958,23 @@ def download_all(self,
             result.
         """
         if isinstance(job_id_or_result_json, str):
-            for url in self.result_urls(job_id_or_result_json, show_progress=False) or []:
-                if url.endswith('zarr'):
-                    raise self.zarr_download_exception
-                yield self.executor.submit(self._download_file, url, directory, overwrite)
+            num_files = 0
+            try:
+                for url in self.result_urls(job_id_or_result_json, show_progress=False) or []:
+                    num_files += 1
+                    if url.endswith('zarr'):
+                        raise self.zarr_download_exception
+                    yield self.executor.submit(self._download_file, url, directory, overwrite)
+            except ProcessingFailedException:
+                print('\nJob Status is failed. There is no file to download.', file=sys.stderr)
+                # set num_files to 1 to bypass the extra error reporting below
+                # as it has already been reported
+                num_files = 1
+
+            if num_files == 0:
```

Comment:
It looks like we could move this check into the try block and not do the workaround on line 972.

Reply:
Refactored code.